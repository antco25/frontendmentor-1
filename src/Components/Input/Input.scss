@import '../Utils/common';

.input {
    display: inline-block;
    margin: 0.5rem 0;
    width: 100%;

    input {
        font-family: inherit;
        outline: none;

        &[type=text], &[type=password] {
            flex-grow: 1;
            border: none;
            color: $neutral-1;
            font-weight: bold;
        }

        &[type=submit] {
            background-color: $primary-2;
            border-color: $primary-2;
            color: $primary-0;
            text-transform: uppercase;
            font-size: 1.125rem;
            box-shadow: 0 -0.1875rem $neutral-3 inset;
        
            &:hover {
                cursor: pointer;
                opacity: 0.8;
            }
        }
    }

    .input-wrapper, input[type=submit] {
        width: 100%;
        border: 0.06rem solid;
        border-radius: 0.25rem;
        padding: 0.75rem 1.25rem;
    }

    .input-wrapper {
        display: flex;
        border-color: $neutral-2;

        &:hover, &.input-focus {
            border-color: $accent-1;
        }

        &.input-error {
            border-color: $primary-1;
            color: $primary-1;

            [type=text], [type=password] {
                color: $primary-1;
            }
        }
    }

    p {
        margin: 0;
        font-size: 0.75rem;
        font-style: italic;
        color: $primary-1;
        text-align: right;
    }

}

