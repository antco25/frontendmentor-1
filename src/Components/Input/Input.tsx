import { ChangeEvent, useState } from 'react'
import './Input.scss';
import iconError from './icon-error.svg'

type InputType = "text" | "submit";

type InputProps = {
    id?: number,
    type?: InputType,
    value?: string,
    isError?: boolean,
    errorMessage?: string,
    onChange?: (e : ChangeEvent<HTMLInputElement>, id: number) => any
}

const Input = ({ id, type, value, isError, errorMessage, onChange}: InputProps) => {

    const [focused, setFocused] = useState(false)

    const onFocus = () => {
        setFocused(true)
    }

    const onBlur = () => {
        setFocused(false)
    }

    if (type === "submit")
        return (
            <div className="input">
                <input type="submit" value={value} />
            </div>
        )

    const wrapperClass = "input-wrapper" + (focused ? " input-focus" : "") + (isError ? " input-error" : "")
    const _onChange = ((onChange != undefined && id != undefined) ? (e : ChangeEvent<HTMLInputElement>) => onChange(e, id) : undefined)

    return (
        <div className="input">
            <div className={wrapperClass}>
                <input type="text" placeholder={value} onFocus={onFocus} onBlur={onBlur} onChange={_onChange} />
                {isError ? <img src={iconError} /> : null}
            </div>
            {isError && errorMessage ? <p>{errorMessage}</p> : null}
        </div>
    )
}

export default Input;