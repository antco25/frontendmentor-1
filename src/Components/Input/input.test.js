import render from 'react-test-renderer'
import Input from './Input'

it('renders correctly', () => {
  const tree = render
    .create(<Input/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with type="submit"', () => {
    const tree = render
      .create(<Input type="submit"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
