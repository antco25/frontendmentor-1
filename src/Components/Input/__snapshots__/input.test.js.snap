// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="input"
>
  <div
    className="input-wrapper"
  >
    <input
      onBlur={[Function]}
      onFocus={[Function]}
      type="text"
    />
  </div>
</div>
`;

exports[`renders correctly with type="submit" 1`] = `
<div
  className="input"
>
  <input
    type="submit"
  />
</div>
`;
